internal/services/containers/kubernetes_nodepool.go

```diff
@@ -1381,6 +1402,11 @@ func FlattenDefaultNodePool(input *[]managedclusters.ManagedClusterAgentPoolProf
 		scaleDownMode = *agentPool.ScaleDownMode
 	}
 
+	snapshotId := ""
+	if agentPool.CreationData != nil && agentPool.CreationData.SourceResourceId != nil {
+		snapshotId = *agentPool.CreationData.SourceResourceId
```

Comment:
We should parse this ID insensitively before setting it to state

Reply:
Thanks! I've updated it.